### PR TITLE
Mapping over an array of objects

### PR DESCRIPTION
**Pull request by D-Pagey**
Please can you have a look at how I have displayed the data I receive from the Wikipedia API? 

I have a feeling the `Article.js` file could be tidied up or made simpler. 

The previous code in the `Article.js` file was just a hard coded article container to visualise how the end result would look.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -32,7 +33,8 @@ class App extends Component {
     fetch(url + this.state.userQuery)
       .then(response => response.json())
       .then(data => this.setState({
-        batch: data.query.search
+        batch: data.query.search,
+        footer: 'footer not-fixed'
```
Comment (pchennz) [inline src/App.js]: if this state will only be used as a className, maybe name it "footerClass".

e.g. during dev, always think - if I revisit this code in 2 years, what can I do to most easily help myself understand what I did?

(more realistic situation - how do I help the person adding changes to this code (most likely someone other than myself) understand what I did)

File: src/App.js [hunk 2]
```diff
@@ -32,7 +33,8 @@ class App extends Component {
     fetch(url + this.state.userQuery)
       .then(response => response.json())
       .then(data => this.setState({
-        batch: data.query.search
+        batch: data.query.search,
```
Comment (pchennz) [inline src/App.js:36]: this state was an object during initialisation, but here it's set to a string?

File: src/components/Articles/Articles.js [hunk 3]
```diff
@@ -3,20 +3,28 @@ import React from 'react';
...
+              rel='noopener noreferrer'>
+              <p className='article-snippet'>
+              {obj.snippet.replace(/<\/?[^>]+>/gi, '')}
+              </p></a>
             </div>
-        </main>
-    )
-}
+        ));
+        } else {
```
Comment (pchennz) [inline src/components/Articles/Articles.js]: extra indentation indentation.

Highly recommend checking out `eslint`, and `airbnb linting standards` - if you manage to integrate both, will help you become a much better programmer

File: src/components/Articles/Articles.js [hunk 4]
```diff
@@ -3,20 +3,28 @@ import React from 'react';
...
+          articleData = null;
+        }
+
+        return <main>{articleData}</main>;
```
Comment (pchennz) [inline src/components/Articles/Articles.js]: currently this component could return `<main>null</main>`

perhaps:
```
if (!props.searchResults || !props.searchResults.length) {
  return null;
}

return <main> .... </main>
```

File: src/components/Footer/Footer.js [hunk 5]
```diff
@@ -2,10 +2,11 @@ import React from 'react';
 
 import './Footer.css';
 
-function Footer() {
+function Footer(props) {
     return (
-        <footer className='footer'>
-            <p>Written and coded by <a href="https://www.freecodecamp.org/d-pagey" 
+        <footer className={props.footerClass}>
```
Comment (pchennz) [inline src/components/Footer/Footer.js:7]: maybe clearer, if Footer accepts "className" prop, then
`<footer className={props.className}>`

Comment (pchennz) [reply src/components/Footer/Footer.js:7]: in parent level, `<Footer className={state.footerClass}>`

**Comment by D-Pagey** [comment]
@pchennz thanks for all the comments. I've amended all the code to take them into account!

**Pull request merged**